src/wallet/wallet.cpp

```diff
@@ -1420,6 +1420,37 @@ void CWallet::transactionAddedToMempool(const CTransactionRef& tx) {
     if (it != mapWallet.end()) {
         RefreshMempoolStatus(it->second, chain());
     }
+
+    const Txid& conflict_txid = tx->GetHash();
+
+    for (const CTxIn& tx_in : tx->vin) {
+        // If the wallet contains no transactions that spend the same prevout, continue
+        if (mapTxSpends.count(tx_in.prevout) == 0) continue;
+
+        std::pair<TxSpends::const_iterator, TxSpends::const_iterator> range;
+        range = mapTxSpends.equal_range(tx_in.prevout);
+
+        // For each wallet transaction spending this prevout...
+        for (TxSpends::const_iterator _it = range.first; _it != range.second; ++_it) {
+            // if this tx is the same as the tx which was just added to the mempool, continue
+            CWalletTx& wallet_tx = mapWallet.at(_it->second);
+            if (wallet_tx.tx->GetHash() == conflict_txid) continue;
+
+            auto try_updating_state = [&](CWalletTx& wtx) EXCLUSIVE_LOCKS_REQUIRED(cs_wallet) {
+                // Add this conflict to the wtx's entry (create one if one does not already exist)
+                wtx.mempool_conflicts.insert(conflict_txid);
+
+                if (wtx.isInactive()) {
+                    wtx.m_state = TxStateInactive{wtx.isAbandoned(), /*mempool_conflicted=*/ true};
```

Comment:
In commit "wallet: track mempool conflicts" (2c867784b2a719c9e701e19e389c3adfa7d7faae)

Can drop this code if `TxStateInactive::mempool_conflicted` member is dropped as suggested above.

Reply:
Done